app/controllers/publishers_controller.rb

```diff
@@ -500,6 +546,21 @@ def should_throttle_resend_auth_email?
     request.env["rack.attack.throttle_data"]["resend_auth_email/publisher_id"][:count] >= THROTTLE_THRESHOLD_RESEND_AUTH_EMAIL
   end
 
+  def should_create_default_currency_card?
+    current_publisher.available_uphold_currencies.exclude?(current_publisher.default_currency) && 
+    current_publisher.possible_uphold_currencies.include?(current_publisher.default_currency)
+  end
+
+  def should_create_bat_card?
+    current_publisher.available_uphold_currencies.exclude?("BAT") && 
+    current_publisher.possible_uphold_currencies.include?("BAT") &&
+    current_publisher.default_currency != "BAT"
+  end
+
+  def should_update_eyeshade_default_currency?
+    current_publisher.eyeshade_default_currency != current_publisher.default_currency
+  end
+
```

Comment:
Would these be more appropriate in the Publisher model?

Reply:
I think that would be preferable, since the logic is wholly based on the model and thus it could be unit tested there.

Reply:
I will move these to the model.